Tests/UnitTests/Identity/CustomerInfoManagerTests.swift

```diff
@@ -415,3 +418,176 @@ class CustomerInfoManagerTests: TestCase {
+        try AvailabilityChecks.iOS13APIAvailableOrSkipTest()
+
+        self.customerInfoManager.cache(customerInfo: self.mockCustomerInfo, appUserID: Self.appUserID)
+
+        let result = try await self.customerInfoManager.customerInfo(appUserID: Self.appUserID,
+                                                                     fetchPolicy: .fromCacheOnly)
+
+        expect(result) == self.mockCustomerInfo
+        expect(self.mockBackend.invokedGetSubscriberData) == false
+    }
+
+    func testCustomerInfoFromCacheOnlyReturnsFromCacheEvenIfExpired() async throws {
+        self.mockDeviceCache.stubbedIsCustomerInfoCacheStale = true
+        self.customerInfoManager.cache(customerInfo: self.mockCustomerInfo, appUserID: Self.appUserID)
+
+        let result = try await self.customerInfoManager.customerInfo(appUserID: Self.appUserID,
+                                                                     fetchPolicy: .fromCacheOnly)
+
+        expect(result) == self.mockCustomerInfo
+        expect(self.mockBackend.invokedGetSubscriberData) == false
+    }
+
+    func testCustomerInfoFromCacheOnlyThrowsWhenNotAvailable() async throws {
+        do {
+            _ = try await self.customerInfoManager.customerInfo(appUserID: Self.appUserID,
+                                                                fetchPolicy: .fromCacheOnly)
+
+            fail("Expected error")
+        } catch BackendError.missingCachedCustomerInfo {
+            // Expected error
+        } catch {
+            fail("Unexpected error: \(error)")
+        }
+
+        expect(self.mockBackend.invokedGetSubscriberData) == false
+    }
+
+    // MARK: - CacheFetchPolicy.cachedOrFetched
+
+    func testCustomerInfoCachedOrFetchedReturnsFromCacheIfAvailable() async throws {
+        self.customerInfoManager.cache(customerInfo: self.mockCustomerInfo, appUserID: Self.appUserID)
+
+        let result = try await self.customerInfoManager.customerInfo(appUserID: Self.appUserID,
+                                                                     fetchPolicy: .cachedOrFetched)
+        expect(result) == self.mockCustomerInfo
+        expect(self.mockBackend.invokedGetSubscriberData) == false
+    }
+
+    func testCustomerInfoCachedOrFetchedReturnsFromCacheAndRefreshesIfStale() async throws {
+        self.mockDeviceCache.stubbedIsCustomerInfoCacheStale = true
+        self.mockBackend.stubbedGetCustomerInfoResult = .success(self.mockRefreshedCustomerInfo)
+
+        self.customerInfoManager.cache(customerInfo: self.mockCustomerInfo, appUserID: Self.appUserID)
+
+        let result = try await self.customerInfoManager.customerInfo(appUserID: Self.appUserID,
+                                                                     fetchPolicy: .cachedOrFetched)
+
+        expect(result) == self.mockCustomerInfo
+        expect(self.mockBackend.invokedGetSubscriberDataCount) == 1
+    }
+
+    func testCustomerInfoCachedOrFetchedFetchesIfNoCache() async throws {
```

Comment:
the only thing I'd add is to have a case for "returns error if cache was stale and new cache can't be fetched" in case of network errors

Reply:
Good idea, added for both `.cachedOrFetched` and `.notStaleCachedOrFetched`.